driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
> If we decide that the Future should not fail then it would be a breaking change in semantic I guess. Could breaks applications that don't notice the change.

Actually it won't break the current API semantic, as currently a `ok` with some write errors happens as a `Future.successful` (but the current result inside may not have all details about the related errors).